### PR TITLE
[SW-1248] getFeaturesCols() should not return the fold column or weight column

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala [hunk 1]
```diff
@@ -52,12 +53,16 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
...
+
     // if this is left empty select
     if (getFeaturesCols().isEmpty) {
-      setFeaturesCols(dataset.columns.filter(n => n.compareToIgnoreCase(getLabelCol()) != 0))
+      val features = dataset.columns.filter(c => excludedCols.forall(e => c.compareToIgnoreCase(e) != 0))
+      setFeaturesCols(features)
     }
 
-    val cols = getFeaturesCols().map(col) ++ Array(col(getLabelCol()))
+    val cols = (getFeaturesCols() ++ excludedCols).map(col)
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala]: 👍

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala [hunk 2]
```diff
@@ -87,16 +92,18 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
     water.DKV.put(trainFrame)
     
     // Train
-    val model: M with H2OModelParams = trainModel(getParams)
+
+    val binaryModel = trainModel(getParams)
+    val model = new H2OMOJOModel(ModelSerializationSupport.getMojoData(binaryModel))
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala]: Here should be `val model: M with H2OModelParams`, unless the methods `setFeaturesCols` etc should not be visible, or am I missing something?

Comment (mn-mikke) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala]: This should be solved by removing ```M``` from the generic parameters.

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2ODeepLearning.scala [hunk 3]
```diff
@@ -17,33 +17,28 @@
...
-import hex.deeplearning.DeepLearning
+import hex.deeplearning.{DeepLearning, DeepLearningModel}
 import hex.deeplearning.DeepLearningModel.DeepLearningParameters
 import hex.schemas.DeepLearningV3.DeepLearningParametersV3
 import org.apache.spark.annotation.Since
-import org.apache.spark.ml.h2o.models.H2OMOJOModel
 import org.apache.spark.ml.h2o.param.H2OAlgoParams
 import org.apache.spark.ml.util._
-import water.support.ModelSerializationSupport
+
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2ODeepLearning.scala]: Nit: extra line?

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OXGBoost.scala [hunk 4]
```diff
@@ -17,31 +17,27 @@
...
+import hex.tree.xgboost.{XGBoost, XGBoostModel}
 import hex.tree.xgboost.XGBoostModel.XGBoostParameters
 import hex.tree.xgboost.XGBoostModel.XGBoostParameters._
 import org.apache.spark.annotation.Since
 import org.apache.spark.ml.h2o.models._
 import org.apache.spark.ml.h2o.param.{EnumParam, H2OAlgoParams}
 import org.apache.spark.ml.param.Params
 import org.apache.spark.ml.util.{Identifiable, MLReadable, MLReader}
-import water.support.ModelSerializationSupport
+
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OXGBoost.scala]: Extra line?

File: ml/src/main/scala/py_sparkling/ml/algos/H2OXGBoost.scala [hunk 5]
```diff
@@ -17,22 +17,19 @@
...
 package py_sparkling.ml.algos
 
-import hex.tree.xgboost.XGBoost
+import hex.tree.xgboost.{XGBoost, XGBoostModel}
 import hex.tree.xgboost.XGBoostModel.XGBoostParameters
 import org.apache.spark.ml.h2o.algos.H2OAlgorithmReader
 import org.apache.spark.ml.util.{MLReadable, MLReader}
 import py_sparkling.ml.models.H2OMOJOModel
-import water.support.ModelSerializationSupport
+
```
Comment (jakubhava) [inline ml/src/main/scala/py_sparkling/ml/algos/H2OXGBoost.scala]: line :)

**Pull request merged**